src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py

```diff
@@ -31,31 +40,88 @@
                                 'provide storage_account value in batchai section of your az configuration file.'
 MSG_CONFIGURE_STORAGE_KEY = 'Please configure Azure Storage account key via AZURE_BATCHAI_STORAGE_KEY or ' \
                             'provide storage_key value in batchai section of your az configuration file.'
+STANDARD_OUTPUT_DIRECTORY_ID = 'stdouterr'
 
 # Placeholders which customer may use in his config file for cluster creation.
 AZURE_BATCHAI_STORAGE_KEY_PLACEHOLDER = '<{0}>'.format(AZURE_BATCHAI_STORAGE_KEY)
 AZURE_BATCHAI_STORAGE_ACCOUNT_PLACEHOLDER = '<{0}>'.format(AZURE_BATCHAI_STORAGE_ACCOUNT)
 
+# Default expiration time for file download URLs.
+DEFAULT_URL_EXPIRY_MIN = 60
+
 # Supported images.
-SUPPORTED_IMAGES = {
-    "ubuntults": models.ImageReference(
+SUPPORTED_IMAGE_ALIASES = {
+    "UbuntuLTS": models.ImageReference(
         publisher='Canonical',
         offer='UbuntuServer',
         sku='16.04-LTS'
     ),
-    "ubuntudsvm": models.ImageReference(
+    "UbuntuDSVM": models.ImageReference(
         publisher='microsoft-ads',
         offer='linux-data-science-vm-ubuntu',
         sku='linuxdsvmubuntu'
     )
 }
 
+# Type of entries reported by list startup files.
+LogFile = collections.namedtuple('LogFile', 'name download_url is_directory size')
+
+logger = get_logger(__name__)
+
+
+def _get_resource_group_location(cli_ctx, resource_group):
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES)
+    return client.resource_groups.get(resource_group).location
```

Comment:
You should use the "get_default_location_from_resource_group" that is included as part of the core.  This will throw a stack trace if the resource group does not exist.

Reply:
thanks, will migrate

Reply:
looks like get_default_location_from_resource_group doesn't fit our needs, or i do not see how it fits

Reply:
Since you're using this method on multiple resource groups, then yes, the stock validator won't work and this is fine.